lib/dry/component/loader.rb

```diff
@@ -3,33 +3,56 @@
 module Dry
   module Component
     class Loader
-      IDENTIFIER_SEPARATOR = '.'.freeze
-      PATH_SEPARATOR = '/'.freeze
+      class Component
+        attr_reader :loader
+        attr_reader :identifier, :path, :file
 
-      attr_reader :identifier, :path, :file
+        def initialize(loader, input)
+          @loader = loader
 
-      def initialize(input)
-        @identifier = input.to_s.gsub(PATH_SEPARATOR, IDENTIFIER_SEPARATOR)
-        @path = input.to_s.gsub(IDENTIFIER_SEPARATOR, PATH_SEPARATOR)
-        @file = "#{path}.rb"
-      end
+          @identifier = input.to_s.gsub(loader.path_separator, loader.namespace_separator)
+          if loader.default_namespace
+            re = /^#{Regexp.escape(loader.default_namespace)}#{Regexp.escape(loader.namespace_separator)}/
+            @identifier = @identifier.gsub(re, '')
+          end
 
-      def namespaces
-        identifier.split(IDENTIFIER_SEPARATOR).map(&:to_sym)
-      end
+          @path = input.to_s.gsub(loader.namespace_separator, loader.path_separator)
```

Comment:
> `tr(from_str, to_str) => new_str`
> Returns a copy of str with the characters in from_str replaced by the corresponding characters in to_str. If to_str > is shorter than from_str, it is padded with its last character in order to maintain the correspondence.

I wonder if this could get us into trouble if the namespace separator was more than one character long? Perhaps gsub is still the safest bet here?